doc/source/configuration.rst

```diff
@@ -0,0 +1,270 @@
+        auth_url: 'https://iam.eu-de.otc.t-systems.com:443/v3'
+      interface: 'public'
+      identity_api_version: 3 # !Important
+      ak: '<AK_VALUE>' # AK/SK pair for access to OBS
+      sk: '<SK_VALUE>'
+    otcsecondproject:
+      profile: otc
+      auth:
+       username: '<USER_NAME>'
+        password: '<PASSWORD>'
+        project_name: '<eu-de_project2>'
+        # or project_id: '<123456_PROJECT_ID2>'
+        user_domain_name: 'OTC00000000001000000xxx'
+        # or user_domain_id: '<123456_DOMAIN_ID2>'
+        auth_url: 'https://iam.eu-de.otc.t-systems.com:443/v3'
+      interface: 'public'
+      identity_api_version: 3 # !Important
+      ak: '<AK_VALUE>' # AK/SK pair for access to OBS
+      sk: '<SK_VALUE>'
+
+Splitting the credentials in clouds.yaml and secure.yaml
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In some scenarios a split of security credentials from the configuration file
+is necessary. The optional file ``secure.yaml`` can be used to store the
+secret which is left out from ``clouds.yaml``:
+
+**clouds.yaml**
+
+.. code-block:: yaml
+
+  clouds:
+    otc:
+      profile: otc
+      auth:
+        username: '<USER_NAME>'
+        project_name: '<eu-de_project>'
+        # or project_id: '<123456_PROJECT_ID>'
+        user_domain_name: 'OTC00000000001000000xxx'
+        # or user_domain_id: '<123456_DOMAIN_ID>'
+        auth_url: 'https://iam.eu-de.otc.t-systems.com:443/v3'
+      interface: 'public'
+      identity_api_version: 3 # !Important
+      ak: '<AK_VALUE>' # AK/SK pair for access to OBS
+      sk: '<SK_VALUE>'
+
+**secure.yaml**
+
+.. code-block:: yaml
+
+  clouds:
+    otc:
+      auth:
+        password: '<PASSWORD>'
+
+.. _environment-variables:
+
+Agency based authorization
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Open Telekom Cloud supports a concept of agencies. One domain delegates access
+to resources to another domain. After trust relationship is established the
+following configuration can be used in ``clouds.yaml``:
+
+.. code-block:: yaml
+
+  clouds:
+    otc:
+      profile: otc
+      auth_type: agency
+      auth:
+        username: '<USER_NAME>'
+        project_name: '<eu-de_project>'
+        # or project_id: '<123456_PROJECT_ID>'
+        user_domain_name: 'OTC00000000001000000xxx'
+        # or user_domain_id: '<123456_DOMAIN_ID>'
+        auth_url: 'https://iam.eu-de.otc.t-systems.com:443/v3'
+        target_domain_id: '<123456_DOMAIN_ID>' # Domain where agency is created
+        # or target_domain_name: '<123456_DOMAIN_NAME'
+        target_agency_name: 'test_agency' # name of the agency
+        target_project_name: '<123456_PROJECT_NAME>' # project scoped operations
+        # or target_project_id: '<123456_PROJECT_ID>'
+        # When target_project_xx is not set - domain scope is selected
+
+Configuration of Environment Variables
+--------------------------------------
+
+Instead of using the clouds.yaml file, environmnt variables can be configured
```

Comment:
Please rephrase sentence here that it somehow does not really advertise to use this way as a preferred one. Maybe something in the form: `historically OpenStack tools are supporting configuration through environment variables...`.

Reply:
rephrased